src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs

```diff
@@ -41,6 +41,24 @@ public static bool IsAttributeName(SyntaxNode node)
         }
 
         /// <summary>
+        /// Returns true if the node is a global attribute.
+        /// </summary>
+        public static bool IsGlobalAttribute(SyntaxNode node)
```

Comment:
this seems something needs to be there. but you probably need to ask @gafter about adding new public compiler API.